Content.Shared/Weapons/Melee/SharedMeleeWeaponSystem.cs

```diff
@@ -463,7 +470,7 @@ private bool AttemptAttack(EntityUid user, EntityUid weaponUid, MeleeWeaponCompo
 
     protected abstract bool InRange(EntityUid user, EntityUid target, float range, ICommonSession? session);
 
-    protected virtual void DoLightAttack(EntityUid user, LightAttackEvent ev, EntityUid meleeUid, MeleeWeaponComponent component, ICommonSession? session)
+    protected virtual void DoLightAttack(EntityUid user, LightAttackEvent ev, EntityUid meleeUid, MeleeWeaponComponent component, ICommonSession? session, EntityUid attacker)
```

Comment:
Can you explain why you are introducing the `EntityUid attacker` here instead of overriding the `EntityUid user` directly? Because it seems to me that the attacker would also be used for `GetDamage` and `GetResistanceBypass` as well, at which point the `attacker` becomes redundant. 

There is also an issue where there's no clear explanation of the difference in `user` and `attacker`; if there is a good reason to keep them separate there should probably be a `///<summary>` attached.